pkg/asset/installconfig/gcp/validation.go

```diff
@@ -73,20 +73,29 @@ func Validate(client API, ic *types.InstallConfig) error {
 	return allErrs.ToAggregate()
 }
 
-func validateInstanceType(fldPath *field.Path, instanceType string, validInstanceTypes []string, arch string) *field.Error {
-	if len(validInstanceTypes) == 0 {
-		return field.InternalError(fldPath, fmt.Errorf("no valid instance types found"))
-	}
-
+func validateInstanceAndDiskType(fldPath *field.Path, diskType, instanceType, arch string) *field.Error {
+	diskTypes := []string{}
+	ok := false
 	if instanceType != "" {
```

Comment:
What about inverting this `if`?
```suggestion
	if instanceType == "" {
	    // nothing to validate, our defaults are already valid
	    return nil
	}
```